src/twisted/internet/base.py

```diff
@@ -449,10 +449,60 @@ def _continueFiring(self, ignored):
                     log.err()
 
 
+@implementer(IReactorPluggableNameResolver, IReactorPluggableResolver)
+class PluggableResolverMixin(object):
```

Comment:
Question: should this be public?

Reply:
(I'm assuming the answer is "yes, since this can be used in third party reactors")

Reply:
Yes.  Making it available for people to use was the point.